### PR TITLE
(bugfix) tagless components error part2 

### LINKED ISSUE #19: You cannot use `attributeBindings` on a tag-less component
I've just migrated my application to ember.js 2.8 and it started throwing error in the console.

`Uncaught Error: Assertion Failed: You cannot use attributeBindings on a tag-less component: <yyyyyy@component:xxxxx::ember382>`

Due to this error, the app doesn't load and shows an empty page.

I've created a small demo app with just ember-hammertime and a tagless component and it still throws the same error: https://github.com/thagul/hammertime-2-8
To verify just run the app and check the console.

### PR DESCRIPTION
**Pull request by eriktrom**
Mimic pre-glimmer2 behavior of attributeBindings

fixes #19

### CODE REVIEW

**Comment by eriktrom** [comment]
@runspired - this worked in 4 or so use cases where other hacks did not. For example, attribute bindings within ember-async-image now work upon deferred loading (as one example https://github.com/runspired/ember-async-image/blob/daf5e88ecc3aafcce9e32394f4848132628e07f0/addon/components/async-image.js#L26)

worth noting I did not mean to re-write the whole mixin - at all - I think there is an upstream bug in Ember TBH and I was about to give up but upon reading the source code saw a line that made me think I could hack around this by modifying the prototype of the mixin from within the constructor - and bam, all edge cases started working - I deleted the cruft but did not back out all the and return the file back to what you are likely more familiar with - let me know if thats an issue and I will

**Comment by eriktrom** [comment]
CI: 
image

File: tests/acceptance/index-test.js [hunk 1]
```diff
@@ -13,9 +13,40 @@ test('visiting /index, ensures we hooked everything up appropriately', function(
...
+  andThen(function() {
+    touchBox().text().indexOf('Evens') > -1;
+    touchBox().text().indexOf('Odds') === -1;
+    assertTouchActionStyle();
+    click(touchBoxEl());
+  });
+
+  andThen(function() {
+    touchBox().text().indexOf('Odds') > -1;
+    touchBox().text().indexOf('Odds') === -1;
```
Comment (eriktrom) [inline tests/acceptance/index-test.js]: thats cute - no assertion - will clean up in morning

Comment (eriktrom) [inline tests/acceptance/index-test.js]: fixed

File: addon/mixins/touch-action.js [hunk 2]
```diff
@@ -2,59 +2,43 @@ import Ember from 'ember';
...
-      newAttributeBindings.push('touchActionStyle:style');
-      set(this, 'attributeBindings', newAttributeBindings);
+    const tagName = this.get('tagName');
+    const hasClick = !!this.get('click');
+    const hasTag = (typeof tagName === 'string' && tagName.length > 0) || (tagName === null && hasClick);
+    if (!hasTag) { return; }
+
+    this.reopen({
+      attributeBindings: ['touchActionStyle:style']
+    });
```
Comment (eriktrom) [inline addon/mixins/touch-action.js]: fwiw - this is the hack that makes this work - i saw why to some extent - but we should notify core team - i dont think regression is near the call site for merged/concatenated/apply mixin code paths in ember source - it seems more disconnected than that (either instantiation of components(lazy or not) or (hopefully not) basic object related (im newb though, just noting what i saw debugging earlier) for example - tagName and element do not exist for elements during some phases of lifecycle, causing often faux 'cant add attribute bindings to a tagless component' error - meaning in most cases, that error is wrong - eventually all components(that dont set tagName to `''` get a tagName, element and id), just not at time of extension via normal reopen of prototype like used to happen - thus my assumption is this hack to works b/c we reopen prototype during/after Class construction, at which point the prototype for the new instance has been assigned to the constructor we're actively invoking - anyway - those are my observation, naive as they may be - but thought i'd share cause i won't remember later :) )

**Comment by eriktrom** [comment]
actually dont merge this - i think i can just take that magic line and paste it into master branch here after dinner - master has some good code i dont want to lose - just fyi

**Comment by runspired** [comment]
@eriktrom wasn't going to merge cause it was a big enough clobber I was going to need to work through it for a while :P a simpler PR is very welcome :)

**Comment by eriktrom** [comment]
yeah i know :) it was after too much hacking and not enough sleep late in the day haha

**Pull request closed without merging**